### PR TITLE
Update explainer

### PR DESCRIPTION
**Pull request by backkem**
This is a suggested update to the explainer to be more in line with the current version of the spec document and to remove some duplicate work (mainly the initial API proposal).

@ibelem I would love your review since you created the original explainer.
@getify please let me know if you feel this sufficiently addresses #20.

***
<a href="https://pr-preview.s3.amazonaws.com/backkem/local-peer-to-peer/pull/26.html" title="Last updated on Dec 13, 2023, 7:12 AM UTC (220b795)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/local-peer-to-peer/26/31cdf70...backkem:220b795.html" title="Last updated on Dec 13, 2023, 7:12 AM UTC (220b795)">Diff</a>

### CODE REVIEW

**Comment by backkem** [comment]
Build fails on a file that isn't changed :/.

**Comment by ibelem** [comment]
Thanks @backkem for updating the explainer and making the API consistent between the explainer and the cg-draft of the spec. Subsequent API updates will be gradually improved, LGTM.

File: EXPLAINER.md [hunk 1]
```diff
@@ -1,57 +1,114 @@
 # Local Peer-to-Peer API Explained
 
-## Introduction
+The Local Peer-to-Peer API aims to give browsers the means to communicate directly, without the aid of a server in the middle. It is designed to enable this communication within the confines of a local communication medium such as the Local Area Network.
 
-The Local Peer-to-Peer API enables mobile and desktop devices to transfer messages or files through close-range communication in a privacy-preserving manner.
+The API aims to strike a balance between creating a powerful new building block for developers and providing a seamless, secure and privacy preserving experience the browser user. As an example: while the API doesn’t provide raw socket access, it does aim to give developers the flexibility to innovate on top by providing a persistent, two-way communication channel with little overhead.
```
Comment (anssiko) [inline EXPLAINER.md:5]:
```suggestion
The API aims to strike a balance between creating a powerful new building block for developers and providing a seamless, secure and privacy preserving experience for the user. As an example: while the API doesn’t provide raw socket access, it does aim to give developers the flexibility to innovate on top by providing a persistent, two-way communication channel with little overhead.
```

File: EXPLAINER.md [hunk 2]
```diff
@@ -1,57 +1,114 @@
...
 
 ## Problem Description
 
-When a user wants to transfer a file from a web site to another device nearby—be it another device the user owns or that of a friend—the user has multiple ways to accomplish this task:
+When a user wants to connect between two devices on the same network, for example to another device nearby—be it another device the user owns or that of a friend—the user has multiple ways to accomplish this task:
 
-1. A centralized cloud service. The file is transferred to a third-party cloud service. This requires a round trip through the internet that consumes network bandwidth and can be slow, costly, or both. The file must be stored on a third-party server, which may not be what the user wants for files that can contain privacy-sensitive information.
+1. _A cloud service_. The web has many ways of connections to a third-party cloud service: HTTP, WebSocket or WebTransport. However, all of these methods require a round trip through the internet. This is inherently dependant on external resources, it consumes network bandwidth and can be slow or costly and has privacy implications in all both the strongest E2E encryption schemes.
```
Comment (anssiko) [inline EXPLAINER.md:23]:
```suggestion
1. _A cloud service_. The web has many ways of connections to a third-party cloud service: HTTP, WebSocket or WebTransport. However, all of these methods require a round trip through the internet. This is inherently dependant on external resources, it consumes network bandwidth and can be slow or costly and has privacy implications in all but the strongest E2E encryption schemes.
```

File: EXPLAINER.md [hunk 3]
```diff
@@ -1,57 +1,114 @@
...
 
-2. A custom WebRTC-based solution. The file is sent between browsers with WebRTC data channels. This solution requires an internet connection for communication with a WebRTC signalling server that is required to establish and manage the connections between the devices.
+2. _A local server_. Many modern Web security measures rely on the presence of naming, signaling and certificate authorities. Local use-cases where these authorities are not readily available have started lagging behind in user experience or are not supported altogether. A local solution involves knowing IPs, Ports and accepting/ignoring a plethora of "Not secure" warnings to get going. This falls short of the user-friendliness that one can expect of -nowadays ubiquitous- cloud services.
```
Comment (anssiko) [inline EXPLAINER.md:25]:
```suggestion
2. _A local server_. Many modern Web security measures rely on the presence of naming, signaling and certificate authorities. Local use-cases where these authorities are not readily available have started lagging behind in user experience or are not supported altogether. A local solution involves knowing IPs, ports and accepting/ignoring a plethora of "Not secure" warnings to get going. This falls short of the user-friendliness that one can expect of—nowadays ubiquitous—cloud services.
```

Maybe we need to use em dash (—) here per the cheatsheet at https://en.wikipedia.org/wiki/Dash
TIL: — looks the same as - using a fixed-width font.

File: EXPLAINER.md [hunk 4]
```diff
@@ -126,104 +183,20 @@ The Local Peer-to-Peer API will cover the following main parts:
...
-
-If the discovery is blocked by the network (e.g., no broadcast-capable medium exists when on an in-flight Wi-Fi service) the underlying implementation may choose to use other technology such as Bluetooth or NFC for discovery and use an appropriate peer to peer technology to transfer the content once the peer has been discovered.
-
-### Transport
-
-Once the discovery process is completed, the content transfer can use a transport layer protocol of choice such as TCP, UDP, QUIC or other, over any link layer protocol available such as Wi-Fi or Wi-Fi Direct. 
+## Non-goals
 
-Similar to the discover phase, supporting data exchange over Bluetooth or NFC could be an optional or fallback implementation as alternative transports.
+- Providing direct access TCP/UDP sockets.
```
Comment (anssiko) [inline EXPLAINER.md:193]:
```suggestion
- Providing direct access to TCP/UDP sockets.
```

**Review by anssiko — APPROVED** [review]
Thank you! Some grammar nits included in the review.

The Bikeshed build failure must be unrelated to this change, related to the major version bump in Bikeshed.

**Comment by getify** [comment]
This all LGTM! Great updates.

**Pull request merged**